libkbfs/disk_block_cache_remote.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	"context"
+	"net"
+
+	"github.com/keybase/client/go/libkb"
+	keybase1 "github.com/keybase/client/go/protocol/keybase1"
+	"github.com/keybase/go-framed-msgpack-rpc/rpc"
+	"github.com/keybase/kbfs/kbfsblock"
+	"github.com/keybase/kbfs/kbfscrypto"
+	kbgitkbfs "github.com/keybase/kbfs/protocol/kbgitkbfs"
+	"github.com/keybase/kbfs/tlf"
+)
+
+type diskBlockCacheRemoteConfig interface {
+	logMaker
+}
+
+// DiskBlockCacheRemote implements a client to access a remote
+// DiskBlockCacheService. It implements the DiskBlockCache interface.
+type DiskBlockCacheRemote struct {
+	conn   net.Conn
+	client kbgitkbfs.DiskBlockCacheClient
+	log    traceLogger
+}
+
+var _ DiskBlockCache = (*DiskBlockCacheRemote)(nil)
+
+// NewDiskBlockCacheRemote creates a new remote disk cache client.
+func NewDiskBlockCacheRemote(kbCtx Context, config diskBlockCacheRemoteConfig) (
+	*DiskBlockCacheRemote, error) {
+	conn, xp, _, err := kbCtx.GetKBFSSocket(true)
+	if err != nil {
+		return nil, err
+	}
+	cli := rpc.NewClient(xp, KBFSErrorUnwrapper{},
+		libkb.LogTagsFromContext)
+	client := kbgitkbfs.DiskBlockCacheClient{Cli: cli}
+
+	return &DiskBlockCacheRemote{
+		conn:   conn,
+		client: client,
+		log:    traceLogger{config.MakeLogger("DBR")},
+	}, nil
+}
+
+// Get implements the DiskBlockCache interface for DiskBlockCacheRemote.
+func (dbcr *DiskBlockCacheRemote) Get(ctx context.Context, tlfID tlf.ID,
+	blockID kbfsblock.ID) (buf []byte,
+	serverHalf kbfscrypto.BlockCryptKeyServerHalf,
+	prefetchStatus PrefetchStatus, err error) {
+	dbcr.log.LazyTrace(ctx, "DiskBlockCacheRemote: Get %s", blockID)
```

Comment:
What would you recommend? This is modeled on `BServerRemote`.

Reply:
Yeah, was specifically asking about the directory.  I think it's supposed to have the 1 for versioning purposes, in case we ever completely revamp it in an incompatible way in the future. All the kbfs-server protocols have the suffix too...

Reply:
I'd recommend `s/DiskBlockCacheRemote: //g`.  No need to repeat the mistakes of bserver_remote.  

Also, just making sure you intend these to be traces, not actual log messages, right?  Because as things are, these will never appear in the logs, only in traces, if tracing happen to be on.

Reply:
Yep, that's correct. The logs are already too chatty.